File: backend/app/Models/StockMovement.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class StockMovement extends Model
{
    protected $fillable = [
        'stock_id', 'production_order_id', 'user_id',
        'type', 'quantity_meter', 'reason',
    ];

    protected $casts = [
        'quantity_meter' => 'float',
    ];

    public function stock()
    {
        return $this->belongsTo(Stock::class);
    }

    public function productionOrder()
    {
        return $this->belongsTo(ProductionOrder::class);
    }

    public function user()
    {
        return $this->belongsTo(User::class);
    }
}

File: backend-dotnet/Models/StockMovement.cs
namespace TexTrack.Api.Models;

public class StockMovement
{
    public long Id { get; set; }
    public long StockId { get; set; }
    public long? ProductionOrderId { get; set; }
    public long? UserId { get; set; }
    public string Type { get; set; } = string.Empty; // in | out
    public decimal QuantityMeter { get; set; }
    public string? Reason { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public Stock? Stock { get; set; }
    public ProductionOrder? ProductionOrder { get; set; }
    public User? User { get; set; }
}
